src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -37,39 +32,40 @@
 DELETE_STARTER_CHUNKS_PER_DOC = 32
 
 
-def connect_to_pinecone():
-    """
-    Connect to Pinecone.
-    This method is called automatically when creating a new KnowledgeBase object.
-    Or when calling `list_canopy_indexes()`.
-    """
-    try:
-        pinecone_init()
-        pinecone_whoami()
-    except Exception as e:
-        raise RuntimeError("Failed to connect to Pinecone. "
-                           "Please check your credentials and try again") from e
+@lru_cache(maxsize=1)
+def _get_global_client() -> Pinecone:
+    return Pinecone()
```

Comment:
I'm not 100% sure this is multi-processing safe, but let's worry about that in the future